### ISSUE TITLE
linter: gitlab format should output repo-relative paths

### ISSUE THREAD

**Issue by ashsearle** [oxc-project/oxc#17999]
### What version of Oxlint are you using?

1.39.0

### What command did you run?

`oxlint --format gitlab`

### What does your `.oxlintrc.json` config file look like?

No `.oxlintrc.json`.

### What happened?

The `location.path` in gitlab formatted output should be relative to the closest git repo root, not the CWD (see [docs](https://docs.gitlab.com/ci/testing/code_quality/#code-quality-report-format)).

> The file containing the code quality violation, expressed as a relative path **in the repository**. Do not prefix with ./.

Steps to reproduce:
```bash
# Create temp folder and git repo:
cd $(mktemp -d)
git init .

# Create subfolder and install oxlint:
mkdir pathFromRepoRoot && cd pathFromRepoRoot
yarn add -DE oxlint

# Create and lint example file:
echo "const unused = 2;" > example.js
yarn oxlint --format gitlab
```

Actual output:
```json
[
  {
    "description": "Variable 'unused' is declared but never used. Unused variables should start with a '_'.",
    "check_name": "eslint(no-unused-vars)",
    "fingerprint": "92988ecf75890d1d",
    "severity": "major",
    "location": {
      "path": "example.js",
      "lines": {
        "begin": 1,
        "end": 1
      }
    }
  }
]
```

Expected output:
```json
[
  {
    "description": "Variable 'unused' is declared but never used. Unused variables should start with a '_'.",
    "check_name": "eslint(no-unused-vars)",
    "fingerprint": "92988ecf75890d1d",
    "severity": "major",
    "location": {
      "path": "pathFromRepoRoot/example.js",
      "lines": {
        "begin": 1,
        "end": 1
      }
    }
  }
]
```

**Comment by connorshea** [comment]
Does it work if you do have an .oxlintrc.json file?

**Comment by ashsearle** [comment]
> Does it work if you do have an .oxlintrc.json file?

No

**Comment by ashsearle** [comment]
A simple workaround is to post-process the output jwith `jq` something like this:
```bash
oxlint --format gitlab | jq --arg prefix $(git rev-parse --show-prefix) '.[] .location.path |= ($prefix + .)'
```

It'd be nice if this hack wasn't required, either by checking git repo root within `oxlint`, or supporting a command-line argument to explicitly set the prefix for the location.path.

**Comment by connorshea** [comment]
ahhh so this occurs when the repo is using something like a `client/` directory, I assume?

**Comment by connorshea** [comment]
Reproduced this with a test repo https://gitlab.com/connorshea/oxlint-issue-repro-template/-/merge_requests/1

The code quality report is uploaded (and worked fine when I have a package.json at the repo root), but when I move everything into a `client/` directory, the report uploads fine but the link here does not actually work:

<img width="931" height="152" alt="Image" src="https://github.com/user-attachments/assets/0b1d628d-d629-40a0-8562-75886b90dd13" />

**Comment by ashsearle** [comment]
> ahhh so this occurs when the repo is using something like a `client/` directory, I assume?

Yes, something like that.

Thanks for progressing the issue and creating a PR.  I hope it's reviewed and merged without too much trouble 🤞

**Comment by connorshea** [comment]
Fix has been merged and will go out with the release on Monday, thanks for the report!
